src/components/ConfigEditor/AzureCredentialsForm.tsx

```diff
@@ -0,0 +1,294 @@
+import React, { ChangeEvent, FunctionComponent, useEffect, useMemo, useReducer, useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { InlineFormLabel, Button, Select, Input } from '@grafana/ui';
+
+import { AzureAuthType, AzureCredentials, isCredentialsComplete } from './AzureCredentials';
+
+export interface Props {
+  managedIdentityEnabled: boolean;
+  oboEnabled: boolean;
+  credentials: AzureCredentials;
+  azureCloudOptions?: SelectableValue[];
+  onCredentialsChange: (updatedCredentials: AzureCredentials) => void;
+  getSubscriptions?: () => Promise<SelectableValue[]>;
+}
+
+export const AzureCredentialsForm: FunctionComponent<Props> = (props: Props) => {
+  const { credentials, azureCloudOptions, onCredentialsChange, getSubscriptions } = props;
+  const hasRequiredFields = isCredentialsComplete(credentials);
+
+  const [subscriptions, setSubscriptions] = useState<Array<SelectableValue<string>>>([]);
+  const [loadSubscriptionsClicked, onLoadSubscriptions] = useReducer((val) => val + 1, 0);
+  useEffect(() => {
+    if (!getSubscriptions || !hasRequiredFields) {
+      updateSubscriptions([]);
+      return;
+    }
+    let canceled = false;
```

Comment:
nit: `cancelled`

Reply:
`canceled` and `cancelled` are equally correct. I usually see `cancel/canceled` but `cancelling/cancellation`.

Example: https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken.canbecanceled?view=net-7.0

Reply:
I will keep as is since it's already used in other datasources.

Reply:
It was removed anyways.